### ISSUE TITLE
Database Setup

### ISSUE THREAD

**Issue by garrettcorn** [iaincollins/next-auth#200]
I'm trying to setup and test next-auth but I'm having a little trouble getting the database to work. This likely isn't an issue with next-auth. I'm sure someone here would be able to set me straight quickly though and I would greatly appreciate the help.

I setup a mongo database at https://mlab.com for testing and added the following:

.env
```
EMAIL_SERVER_USER=xxx@xxx
EMAIL_SERVER_PASSWORD=emailserverpassword
EMAIL_SERVER_HOST=emailhost
EMAIL_SERVER_PORT=25
EMAIL_FROM=NextAuth <noreply@example.com>
DATABASE_URL=mongodb://<username>:<password>@xxxxxx.mlab.com:57268/<dbname>?synchronize=true
```

`<username>`,`<password>`, and `<dbname>` - username is a database user, database user password, and the database name respectively.

I then proceeded to add the following.

/pages/api/[...nextauth].js
```
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  site: process.env.SITE || 'http://localhost:3000',
  providers: [
    Providers.Email({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: process.env.EMAIL_SERVER_PORT,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD
        }
      },
      from: process.env.EMAIL_FROM
    }),
  ],
  database: process.env.DATABASE_URL
}

export default (req, res) => NextAuth(req, res, options)
```

In my browser I am able to get the the email sign in page, but when I attempt to submit my email I am met with the following:

```
Sign in failed

Unable to send email to your address.

Try signing in again with a different account.
```

The node output message makes it look like it is trying to connect to the database on localhost, so I feel like something is wrong with the db connection setup.

```
ADAPTER_CONNECTION_ERROR MongoNetworkError: failed to connect to server [127.0.0.1:27017] on first connect [Error: connect ECONNREFUSED 127.0.0.1:27017
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  name: 'MongoNetworkError'
}]
```

Thanks ahead of time to anyone who is able to help.

Regards,
Garrett Cornwell

**Comment by iaincollins** [comment]
Hmm that is odd, your setup looks good to me.

I have an mlab account so I'll give it a try with that and see if anything funky is up.

You might also want to experiment be with passing the db details as an object, just to see if that helps, but what you are doing should work.

BTW I think mlab is actually shutting down instances in a couple of weeks (since they got bought by Atlas), but I could be wrong - just a heads up!

**Comment by garrettcorn** [comment]
It is working when provided as an object! I added a log output to make sure the environment variable was being picked up and it looked good to me, but isn't working.

```
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  site: process.env.SITE || 'http://localhost:3000',
  providers: [
    Providers.Email({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: process.env.EMAIL_SERVER_PORT,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD
        }
      },
      from: process.env.EMAIL_FROM
    }),
  ],
  // database: process.env.DATABASE_URL
  database: {
    type: 'mongodb',
    host: "aaa.mlab.com",
    port: 57268,
    username: "bbb",
    password: "ccc",
    database: "ddd",
    synchronize: true
  }
}

export default (req, res) => {
  console.log(process.env.DATABASE_URL)
  NextAuth(req, res, options)
}
```

node log output is

```
mongodb://bbb:ccc@aaa.mlab.com:57268/ddd?synchronize=true
```

**Comment by garrettcorn** [comment]
Setting the database with a string literal isn't working for me either. Looks like it is trying to connect to localhost again.

```
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  site: process.env.SITE || 'http://localhost:3000',
  providers: [
    Providers.Email({
      server: {
        host: process.env.EMAIL_SERVER_HOST,
        port: process.env.EMAIL_SERVER_PORT,
        auth: {
          user: process.env.EMAIL_SERVER_USER,
          pass: process.env.EMAIL_SERVER_PASSWORD
        }
      },
      from: process.env.EMAIL_FROM
    }),
  ],
  // NOT WORKING
  // database: process.env.DATABASE_URL
  // WORKING
  // database: {
  //   type: 'mongodb',
  //   host: "aaa.mlab.com",
  //   port: 57268,
  //   username: "bbb",
  //   password: "ccc",
  //   database: "ddd",
  //   synchronize: true
  // }
  // NOT WORKING
  database: 'mongodb://bbb:ccc@aaa.mlab.com:57268/ddd?synchronize=true'
}

export default (req, res) => {
  console.log(process.env.DATABASE_URL)
  NextAuth(req, res, options)
}
```

Node output

```
ADAPTER_CONNECTION_ERROR MongoNetworkError: failed to connect to server [127.0.0.1:57268] on first connect [Error: connect ECONNREFUSED 127.0.0.1:57268
```

**Comment by iaincollins** [comment]
Ah, thank you! I found the bug that is causing the hostname not to be passed correctly when it's provided as a string. Will publish a fix in the next hour or so and let you know.

Thanks for the detailed report and info which made this easy to track down!

**Comment by iaincollins** [comment]
Fixed published in `next-auth@2.0.0-beta.60`
